### ISSUE TITLE
file upload API param relative_path not working

### ISSUE THREAD

**Issue by Frolanta** [haiwen/seafile#1977]
It seems the relative_path param is not working.
I tried with CURL (following the documentation for api v2.1) and my own nodejs client and I can’t manage to make it work.

But since it’s working when using the web interface I was curious.
So after a network inspection of the request I can see the it use a different URL: **upload-aj** instead of **upload-api** so I tried to replace this in the URL return by the **upload-link** and it’s working great, with CURL and in node.

Not really useful but that’s how I fixed the issue: 

```
export const uploadFile = (repoId, dirName, srcPath, fileName, relativePath = false) => {
  let name = 'uploadFile';
  return getFileUploadLink(repoId, dirName)
    .then(uploadLink => {

      let form = new FormData();
      form.append('file', fs.createReadStream(srcPath), { filename: fileName });
      form.append('parent_dir', dirName);

      if (relativePath) {
        uploadLink = uploadLink.replace('upload-api', 'upload-aj');
        form.append('relative_path', relativePath);
      }
      uploadLink = `${uploadLink}?ret-json=1`;

      return clientSeafile.callPure('POST', uploadLink, form, form.getHeaders())
    })
    .then(res => {
      successLog(name, res);
      return res;
    })
    .catch(err => {
      failLog(name, err);
      throw new Error(err);
    })
};

```
Just added this line `uploadLink = uploadLink.replace('upload-api', 'upload-aj');` it’s working for now but it’s ugly 😕

With the **upload-api** URL the file is uploaded but to the **parent_dir** and not **parent_dir/relative_path**.

I’m using seafile server 6.1.1 running on Linux 64bits

**Comment by mauriciobaeza** [comment]
Confirm this bug. I'm using Server Version: 6.2.3 in Centos7 x64

Thanks.

**Comment by galaxyyao** [comment]
Confirmed not fixed in 6.2.5 in CentOS 7 x64
Thanks for the workaround, but still hope this bug to be fixed.

**Comment by imwhatiam** [comment]
Hello, I tested it on our demo site, https://demo.seafile.top (6.3.7 pro),  and it works fine.

![image](https://user-images.githubusercontent.com/3793639/47707700-6f30a000-dc67-11e8-8b96-f469d6eb7c53.png)
![image](https://user-images.githubusercontent.com/3793639/47707822-b6b72c00-dc67-11e8-8e26-a66f3af9236d.png)
